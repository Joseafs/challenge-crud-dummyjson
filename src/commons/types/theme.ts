import { PropsThemePaletteKeys } from '~/theme/config/types';

export type PropsThemeColor = {
  color?: PropsThemePaletteKeys;
};
