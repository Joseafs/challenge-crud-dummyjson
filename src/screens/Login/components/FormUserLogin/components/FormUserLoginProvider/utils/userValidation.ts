import * as Yup from 'yup';

const wordRequiredLogin = (value: string) => `Por favor, insira ${value}!`;

export const userLoginValidationShema = Yup.object().shape({
  password: Yup.string().required(wordRequiredLogin('sua senha')),
  username: Yup.string().required(wordRequiredLogin('seu usu√°rio')),
});
