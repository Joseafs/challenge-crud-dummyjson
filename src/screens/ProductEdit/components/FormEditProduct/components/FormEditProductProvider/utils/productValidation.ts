import * as Yup from 'yup';

const wordTooShort = 'Muito curto !';
const wordTooLong = 'Muito longo !';
const wordPriceCantBeZero = (value: string) => `O "${value}" não pode ser zero`;
const wordRequiredByValue = (value: string) => `Campo "${value}" é obrigatório.`;

export const productValidationShema = Yup.object().shape({
  brand: Yup.string().min(3, wordTooShort).max(30, wordTooLong).required(wordRequiredByValue('Marca')),
  description: Yup.string().min(3, wordTooShort).max(200, wordTooLong).required(wordRequiredByValue('Descrição')),
  title: Yup.string().min(3, wordTooShort).max(50, wordTooLong).required(wordRequiredByValue('Título')),

  category: Yup.string().required('Por favor, selecione uma categoria.'),
  thumbnail: Yup.string().url().required('Insira uma imagem e a promova a Capa'),

  discountPercentage: Yup.number().min(0).required(wordPriceCantBeZero('Porcentagem de Desconto')),
  price: Yup.number().min(0).required(wordPriceCantBeZero('Preço')),
  rating: Yup.number().min(0).required(wordRequiredByValue('Avaliação')),
  stock: Yup.number().min(0).required(wordRequiredByValue('Estoque')),

  urlImage: Yup.string().when('images', {
    is: (value: []) => value && value.length > 0,
    then: (schema) => schema.required('Por gentileza, insira uma URL de imagem válida.'),
  }),
});
