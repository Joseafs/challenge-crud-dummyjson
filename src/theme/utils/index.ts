export { useMediaQuery } from './useMediaQuery';
export { useThemeColor } from './useThemeColor';
export { useSpaceTheme } from './useThemeSpace';
export { withThemeProvider } from './withThemeProvider';
