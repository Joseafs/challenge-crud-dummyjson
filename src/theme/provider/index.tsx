'use client';

import emotionIsPropValid from '@emotion/is-prop-valid';
import { PropsWithChildren } from 'react';
import { StyleSheetManager, ThemeProvider as StyledComponentsThemeProvider } from 'styled-components';

import theme from '~/theme/config';
import { GlobalStyles } from '~/theme/config/globals';

export const ThemeProvider = ({ children }: PropsWithChildren) => (
  <StyleSheetManager
    enableVendorPrefixes
    shouldForwardProp={(propName, elementToBeRendered) => {
      return typeof elementToBeRendered === 'string' ? emotionIsPropValid(propName) : true;
    }}
  >
    <GlobalStyles theme={theme} />
    <StyledComponentsThemeProvider theme={theme}>{children}</StyledComponentsThemeProvider>
  </StyleSheetManager>
);
