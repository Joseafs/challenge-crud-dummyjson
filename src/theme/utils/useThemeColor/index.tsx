import { useTheme } from 'styled-components';

import { PropsThemeColorTypesKeys, PropsThemePaletteKeys } from '~/theme/config/types';

export const useThemeColor = (color?: PropsThemePaletteKeys, type?: PropsThemeColorTypesKeys) => {
  const theme = useTheme();

  if (color && !theme.palette[color]) {
    return theme.palette.common.black;
  }
  if (type && color) {
    return theme.palette[color][type];
  }
  if (color) {
    return theme.palette[color]?.main || '';
  }

  return theme.palette.common.black;
};
