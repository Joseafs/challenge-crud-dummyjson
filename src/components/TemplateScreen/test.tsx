import { render } from '@testing-library/react';

import { withThemeProvider } from '~/theme/utils';
import { snapshotYamamotoDescription } from '~/utils/mocks/snapshotMockValues';

import { TemplateScreen } from '.';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockReturnValue({
    back: jest.fn(),
  }),
}));

describe('TemplateScreen', () => {
  test('Should match snapshot', () => {
    const { container } = render(withThemeProvider(<TemplateScreen>{snapshotYamamotoDescription}</TemplateScreen>));
    expect(container).toMatchSnapshot();
  });

  // TODO: after add header -> test header only
  // TODO: test body only
  // TODO: after add footer -> test footer only
});
