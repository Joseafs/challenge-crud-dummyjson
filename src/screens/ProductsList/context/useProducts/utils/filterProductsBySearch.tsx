import { ProductData } from '~/services/products/getProduct/types';
import { escapeRegExp } from '~/utils/escapeRegExp';

export const filterProductsBySearch = (products: ProductData[], searchText: string): ProductData[] => {
  if (!searchText) {
    return products;
  }

  const escapedSearchText = escapeRegExp(searchText);
  const regex = new RegExp(escapedSearchText, 'i');

  return products.filter((product) => {
    return regex.test(product.title) || regex.test(product.brand);
  });
};
