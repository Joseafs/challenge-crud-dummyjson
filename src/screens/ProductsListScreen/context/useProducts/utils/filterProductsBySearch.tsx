import { Product } from '~/services/products/getProducts/types';
import { escapeRegExp } from '~/utils/escapeRegExp';

export const filterProductsBySearch = (products: Product[], searchText: string): Product[] => {
  if (!searchText) {
    return products;
  }

  const escapedSearchText = escapeRegExp(searchText);
  const regex = new RegExp(escapedSearchText, 'i');

  return products.filter((product) => {
    return regex.test(product.title) || regex.test(product.brand);
  });
};
