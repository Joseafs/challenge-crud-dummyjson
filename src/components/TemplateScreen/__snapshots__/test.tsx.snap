// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateScreen Should match snapshot 1`] = `
<div>
  <div
    class="sc-36588e0e-0 jTkvOt"
  >
    <header
      class="sc-36588e0e-1 cvtwAa"
    >
      <button
        class="sc-999c1705-0 gHEbJm sc-7d50a5a3-0 kXJZYA"
        title="Voltar"
      >
        <svg
          class="sc-7d50a5a3-1 jTrtZF"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fill-rule="evenodd"
          />
        </svg>
        Voltar
      </button>
      <div>
        Lorem user ?
      </div>
    </header>
    <main
      class="sc-36588e0e-2 ktNfYw"
    >
      Yamamoto, the Captain-Commander of the Gotei 13, wields the fearsome Zanpakuto Ryujin Jakka. His fiery power and unwavering resolve make him a legend in Soul Society.
    </main>
  </div>
</div>
`;
